model/channel_shared.go

```diff
@@ -0,0 +1,166 @@
+	}
+
+	if sc.UpdateAt == 0 {
+		return NewAppError("Channel.IsValid", "model.channel.is_valid.update_at.app_error", nil, "id="+sc.ChannelId, http.StatusBadRequest)
+	}
+
+	if utf8.RuneCountInString(sc.ShareDisplayName) > CHANNEL_DISPLAY_NAME_MAX_RUNES {
+		return NewAppError("Channel.IsValid", "model.channel.is_valid.display_name.app_error", nil, "id="+sc.ChannelId, http.StatusBadRequest)
+	}
+
+	if !IsValidChannelIdentifier(sc.ShareName) {
+		return NewAppError("Channel.IsValid", "model.channel.is_valid.2_or_more.app_error", nil, "id="+sc.ChannelId, http.StatusBadRequest)
+	}
+
+	if utf8.RuneCountInString(sc.ShareHeader) > CHANNEL_HEADER_MAX_RUNES {
+		return NewAppError("Channel.IsValid", "model.channel.is_valid.header.app_error", nil, "id="+sc.ChannelId, http.StatusBadRequest)
+	}
+
+	if utf8.RuneCountInString(sc.SharePurpose) > CHANNEL_PURPOSE_MAX_RUNES {
+		return NewAppError("Channel.IsValid", "model.channel.is_valid.purpose.app_error", nil, "id="+sc.ChannelId, http.StatusBadRequest)
+	}
+
+	if !IsValidId(sc.CreatorId) {
+		return NewAppError("Channel.IsValid", "model.channel.is_valid.creator_id.app_error", nil, "CreatorId="+sc.CreatorId, http.StatusBadRequest)
+	}
+
+	if !sc.Home {
+		if !IsValidId(sc.RemoteClusterId) {
+			return NewAppError("Channel.IsValid", "model.channel.is_valid.id.app_error", nil, "RemoteClusterId="+sc.RemoteClusterId, http.StatusBadRequest)
+		}
+		if !IsValidId(sc.Token) {
+			return NewAppError("Channel.IsValid", "model.channel.is_valid.id.app_error", nil, "Token", http.StatusBadRequest)
+		}
+	}
+	return nil
+}
+
+func (sc *SharedChannel) PreSave() {
+	sc.ShareName = SanitizeUnicode(sc.ShareName)
+	sc.ShareDisplayName = SanitizeUnicode(sc.ShareDisplayName)
+
+	sc.CreateAt = GetMillis()
+	sc.UpdateAt = sc.CreateAt
+}
+
+func (sc *SharedChannel) PreUpdate() {
+	sc.UpdateAt = GetMillis()
+	sc.ShareName = SanitizeUnicode(sc.ShareName)
+	sc.ShareDisplayName = SanitizeUnicode(sc.ShareDisplayName)
+}
+
+// SharedChannelRemote represents a remote cluster that has been invited
+// to a shared channel.
+type SharedChannelRemote struct {
+	Id                string `json:"id"`
+	ChannelId         string `json:"channel_id"`
+	Token             string `json:"token"`
+	Description       string `json:"description"`
+	CreatorId         string `json:"creator_id"`
+	CreateAt          int64  `json:"create_at"`
+	UpdateAt          int64  `json:"update_at"`
+	IsInviteAccepted  bool   `json:"is_invite_accepted"`
+	IsInviteConfirmed bool   `json:"is_invite_confirmed"`
+	RemoteClusterId   string `json:"remote_cluster_id"`
+}
+
+func (sc *SharedChannelRemote) ToJson() string {
+	b, _ := json.Marshal(sc)
+	return string(b)
+}
+
+func SharedChannelRemoteFromJson(data io.Reader) (*SharedChannelRemote, error) {
+	var sc *SharedChannelRemote
+	err := json.NewDecoder(data).Decode(&sc)
+	return sc, err
+}
+
+func (sc *SharedChannelRemote) IsValid() *AppError {
```

Comment:
Maybe here we can check that the Description is not longer than `64` (the sql limit set later in code).

Also maybe check that the token is valid based on the values IsInviteAccepted and IsInviteConfirmed.

Reply:
Thanks @ashishbhate.  Description length check added.  https://github.com/mattermost/mattermost-server/pull/16065/commits/5b182d5c65ca237d5bdcdb5ef13bf9bf1b293079